### PR TITLE
Feature/nex 1135/sql iterators types support

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/NEX-1135

* Allows specifying limits
* Allows more flexible parametrization by explicit param types specification  ( compatibility with PDO improved ) 
* get rid of redundent PHPDoc

### CODE REVIEW

**Comment by siwane** [comment]
Nice cleaning 👍

File: common/persistence/sql/class.QueryIterator.php [hunk 1]
```diff
@@ -14,99 +14,78 @@
...
     /**
      * @var common_persistence_SqlPersistence
      */
     private $persistence;
-    
-    /**
-     * Query to iterator over
-     *
-     * @var string
-     */
```
Comment (bartlomiejmarszal) [inline common/persistence/sql/class.QueryIterator.php:40]: Why did you remove them?

File: common/persistence/sql/class.QueryIterator.php [hunk 2]
```diff
@@ -14,99 +14,78 @@
...
-    public function __construct(common_persistence_SqlPersistence $persistence, $query, $params = [])
-    {
+    private $cache;
+    private $types;
+    private $limit;
+
+    public function __construct(
+        common_persistence_SqlPersistence $persistence,
+        $query,
+        $params = [],
```
Comment (gabrielfs7) [inline common/persistence/sql/class.QueryIterator.php:58]:
```suggestion
        string $query,
        array $params = [],
```

Now we can add type hint :)

File: common/persistence/sql/class.QueryIterator.php (hunk 1 above)
Comment (gabrielfs7) [reply common/persistence/sql/class.QueryIterator.php:40]: @boajer  As we do not have php7.4 we still need `@var` php doc for attributes. When you put it back, I would remove the useless comments and fix the `$params` which is an array, not a string.
